File: Exercises/Ex5/Matteo/random_generator.cpp
#include <iostream>
#include <TRandom3.h>
#include <TFile.h>
#include <TTree.h>

void random_generator() {
    TRandom3 rng(0);

    TFile file("random_numbers.root", "RECREATE");

    TTree tree("random_tree", "Tree with random numbers");

    float number;

    tree.Branch("number", &number);

    const int N = 1000;
    for (int i = 0; i < N; ++i) {
        number = rng.Gaus(0, 1); 
        tree.Fill();
    }

    tree.Write();

    file.Close();
}

int main() {
    random_generator();
    return 0;
}

File: Exercises/Ex6/Matteo/random_generator.py
import ROOT

def random_generator():
    rng = ROOT.TRandom3(0)

    file = ROOT.TFile("random_numbers.root", "RECREATE")

    tree = ROOT.TTree("random_tree", "Tree with random numbers")

    number = ROOT.std.vector('float')()
    tree.Branch("number", number)

    N = 1000
    for i in range(N):
        num = rng.Gaus(0, 1)
        number.push_back(num)

    tree.Fill()

    tree.Write()

    file.Close()

if __name__ == "__main__":
    random_generator()
